src/libraries/Microsoft.Extensions.Configuration.CommandLine/src/CommandLineConfigurationProvider.cs

```diff
@@ -39,96 +39,159 @@ public CommandLineConfigurationProvider(IEnumerable<string> args, IDictionary<st
-                        }
-
-                        value = enumerator.Current;
-                    }
-                    else
-                    {
-                        string keySegment = currentArg.Substring(0, separator);
-
-                        // If the switch is a key in given switch mappings, interpret it
-                        if (_switchMappings != null && _switchMappings.TryGetValue(keySegment, out string mappedKeySegment))
-                        {
-                            key = mappedKeySegment;
-                        }
-                        // If the switch starts with a single "-" and it isn't in given mappings , it is an invalid usage
-                        else if (keyStartIndex == 1)
-                        {
-                            throw new FormatException(SR.Format(SR.Error_ShortSwitchNotDefined, currentArg));
-                        }
-                        // Otherwise, use the switch name directly as a key
-                        else
-                        {
-                            key = currentArg.Substring(keyStartIndex, separator - keyStartIndex);
-                        }
-
-                        value = currentArg.Substring(separator + 1);
-                    }
+            Data = data;
+        }
+
+        /// <summary>
+        /// Reconcile two consecutive arguments into a key-value pair for the first.
+        /// </summary>
+        /// <remarks>
+        /// Helper function to reduce repeated code in <see cref="Load"/> method.
+        /// </remarks>
+        /// <param name="previousArg">The first string argument</param>
+        /// <param name="currentArg">The second string argument</param>
+        /// <param name="pair">
+        /// A properly resolved configuration key-value pair, or null if previous argument is invalid.
+        /// </param>
+        /// <returns>
+        /// True if the args can be resolved to a proper configuration key-value pair.
+        /// </returns>
+        private bool TryProcessArgs(string previousArg, string currentArg, out (string key, string value) pair)
+        {
+            string key, value;
+            int keyStartIndex = 0;
 
-                    // Override value when key is duplicated. So we always have the last argument win.
-                    data[key] = value;
+            if (previousArg.StartsWith("--"))
+            {
+                keyStartIndex = 2;
+            }
+            else if (previousArg.StartsWith("-"))
+            {
+                keyStartIndex = 1;
+            }
+            else if (previousArg.StartsWith("/"))
+            {
+                // "/SomeSwitch" is equivalent to "--SomeSwitch" when interpreting switch mappings
+                // So we do a conversion to simplify later processing
+                previousArg = $"--{previousArg.Substring(1)}";
+                keyStartIndex = 2;
+            }
+
+            int separator = previousArg.IndexOf('=');
```

Comment:
nit, rename to `separatorIndex`?

Reply:
Sure. Many of these variable names predate my changes, and I tried to update only what I needed to update. But I agree that's a better name and I changed it.